service/history/transferQueueActiveTaskExecutor_test.go

```diff
@@ -722,6 +723,96 @@ func (s *transferQueueActiveTaskExecutorSuite) TestProcessCloseExecution_HasPare
 	s.Nil(err)
 }
 
+func (s *transferQueueActiveTaskExecutorSuite) TestProcessCloseExecution_CanSkipVisibilityArchival() {
+	for _, skipVisibilityArchival := range []bool{
+		false,
+		true,
+	} {
+		s.Run(fmt.Sprintf("CanSkipVisibilityArchival=%v", skipVisibilityArchival), func() {
+			execution := commonpb.WorkflowExecution{
+				WorkflowId: "some random workflow ID",
+				RunId:      uuid.New(),
+			}
+			workflowType := "some random workflow type"
+			taskQueueName := "some random task queue"
+
+			mutableState := workflow.TestGlobalMutableState(
+				s.mockShard,
+				s.mockShard.GetEventsCache(),
+				s.logger,
+				s.version,
+				execution.GetRunId(),
+			)
+			_, err := mutableState.AddWorkflowExecutionStartedEvent(
+				execution,
+				&historyservice.StartWorkflowExecutionRequest{
+					Attempt:     1,
+					NamespaceId: s.namespaceID.String(),
+					StartRequest: &workflowservice.StartWorkflowExecutionRequest{
+						WorkflowType:             &commonpb.WorkflowType{Name: workflowType},
+						TaskQueue:                &taskqueuepb.TaskQueue{Name: taskQueueName},
+						WorkflowExecutionTimeout: timestamp.DurationPtr(2 * time.Second),
+						WorkflowTaskTimeout:      timestamp.DurationPtr(1 * time.Second),
+					},
+				},
+			)
+			s.Nil(err)
+
+			wt := addWorkflowTaskScheduledEvent(mutableState)
+			event := addWorkflowTaskStartedEvent(mutableState, wt.ScheduledEventID, taskQueueName, uuid.New())
+			wt.StartedEventID = event.GetEventId()
+			event = addWorkflowTaskCompletedEvent(
+				mutableState,
+				wt.ScheduledEventID,
+				wt.StartedEventID,
+				"some random identity",
+			)
+
+			taskID := int64(59)
+			event = addCompleteWorkflowEvent(mutableState, event.GetEventId(), nil)
+
+			transferTask := &tasks.CloseExecutionTask{
+				WorkflowKey: definition.NewWorkflowKey(
+					s.namespaceID.String(),
+					execution.GetWorkflowId(),
+					execution.GetRunId(),
+				),
+				Version:                   s.version,
+				TaskID:                    taskID,
+				VisibilityTimestamp:       time.Now().UTC(),
+				CanSkipVisibilityArchival: skipVisibilityArchival,
+			}
+
+			persistenceMutableState := s.createPersistenceMutableState(
+				mutableState,
+				event.GetEventId(),
+				event.GetVersion(),
+			)
+			s.mockExecutionMgr.EXPECT().GetWorkflowExecution(gomock.Any(), gomock.Any()).
+				Return(&persistence.GetWorkflowExecutionResponse{State: persistenceMutableState}, nil)
+			if !skipVisibilityArchival {
+				s.mockArchivalMetadata.EXPECT().GetVisibilityConfig().
+					Return(archiver.NewArchivalConfig(
+						"enabled",
+						dc.GetStringPropertyFn("enabled"),
+						dc.GetBoolPropertyFn(true),
+						"disabled",
+						"random URI",
+					))
+				s.mockArchivalClient.EXPECT().Archive(gomock.Any(), gomock.Any()).Return(nil, nil)
```

Comment:
Should this expect 0 call if the canSkipArchival flag is true?

Reply:
Yeah definitely. This is already covered though because we only run this expectation if `skipVisibilityArchival` is false:
```go
if !skipVisibilityArchival {
	s.mockArchivalMetadata.EXPECT().GetVisibilityConfig().
		Return(archiver.NewArchivalConfig(
			"enabled",
			dc.GetStringPropertyFn("enabled"),
			dc.GetBoolPropertyFn(true),
			"disabled",
			"random URI",
		))
	s.mockArchivalClient.EXPECT().Archive(gomock.Any(), gomock.Any()).Return(nil, nil)
	s.mockSearchAttributesProvider.EXPECT().GetSearchAttributes(gomock.Any(), false)
}
```